notebooks/py/allen_smFISH.py

```diff
@@ -59,71 +55,70 @@
 # EPY: END code
 
 # EPY: START markdown
 #For bandpass, there's a point where things get weird, at `c == 0; z <= 14`. In that range the images look mostly like noise. However, _above_ that, they look great + background subtracted! The later stages of the pipeline appear robust to this, though, as no spots are called for the noisy sections.
 # EPY: END markdown
 
 # EPY: START code
-# I wasn't sure if this clipping was supposed to be by volume or tile. I've done tile here, but it can be easily
-# switched to volume.
+# I wasn't sure if this clipping was supposed to be by volume or tile. I've done tile here, but it can be switched
 clip = Filter.Clip(p_min=10, p_max=100, is_volume=False)
 clip.run(primary_image, verbose=True, in_place=True)
 # EPY: END code
 
+# EPY: START code
+from showit import image
+image(primary_image.get_slice({Indices.CH.value: 0, Indices.Z.value: 17})[0][0, :, :], size=20, clim=(0, 0.004))
+# EPY: END code
+
 # EPY: START code
 sigma=(1, 0, 0)  # filter only in z, do nothing in x, y
-glp = Filter.GaussianLowPass(sigma=sigma, is_volume=True, verbose=True)
-glp.run(primary_image, in_place=True)
+glp = Filter.GaussianLowPass(sigma=sigma, is_volume=True)
+glp.run(primary_image, in_place=True, verbose=True)
 # EPY: END code
 
 # EPY: START markdown
 #Below, because spot finding is so slow when single-plex, we'll pilot this on a max projection to show that the parameters work. Here's what trackpy.locate, which we wrap, produces for a z-projection of channel 1. To do use our plotting methods on z-projections we have to expose some of the starfish internals, which will be improved upon.
 # EPY: END markdown
 
-# EPY: START code
-from trackpy import locate
-
-# grab a section from the tensor.
-ch1 = primary_image.max_proj(Indices.Z)[0, 1]
-
-results = locate(ch1, diameter=3, minmass=250, maxsize=3, separation=5, preprocess=False, percentile=10)
-results.columns = ['x', 'y', 'intensity', 'radius', 'eccentricity', 'signal', 'raw_mass', 'ep']
-# EPY: END code
-
-# EPY: START code
-results.head()
-# EPY: END code
+# EPY: START markdown
+#Below spot finding is on the _volumes_ for each channel. This will take about `11m30s`
+# EPY: END markdown
 
 # EPY: START code
-#TODO ambrosejcarr: Showing spots is broken right now.
+plt.hist(np.ravel(primary_image.get_slice({Indices.CH.value: 0, Indices.Z.value: 17})[0]), bins=25, log=True);
 # EPY: END code
 
 # EPY: START code
-# plot the z-projection
-f, ax = plt.subplots(figsize=(20, 20))
-ax.imshow(ch1, vmin=15, vmax=52, cmap=plt.cm.gray)
-
-# draw called spots on top as red circles
-# scale radius plots the red circle at scale_radius * spot radius
-# image._show_spots(results, ax=plt.gca(), scale_radius=7);
+# if max radius is 3, and mean intensity is about 0.006, and we can expect spots to have around pi*r^2 integrated intensity... 
+# try half-maximal radius * mean intensity
+(np.pi * 1.5 ** 2) * 0.006
```

Comment:
what's this for?

Reply:
Don't remember. Gone now.

Reply:
Added explanation. Thanks for the catch.